gpu-enabler/src/main/scala/com/ibm/gpuenabler/CUDADSUtils.scala

```diff
@@ -310,13 +345,14 @@ object CUDADSImplicits {
     * improves performance.
     */
     def getPartSizes: Broadcast[Map[Int, Int]] = {
-      val logPlan = ds.queryExecution.logical match {
+      val execPlan = ds.queryExecution.executedPlan
+      val logPlan = ds.queryExecution.optimizedPlan match {
         case SerializeFromObject(_, lp) => lp
-        case _ => ds.queryExecution.logical
+        case _ => ds.queryExecution.optimizedPlan
```

Comment:
'optimizedPlan' should be reverted back to 'logical' for performance implications.